l10n_be_cooperator_portal_national_number/controllers/portal.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from collections import namedtuple
+
+from odoo import _
+from odoo.http import request, route
+
+from odoo.addons.account.controllers.portal import PortalAccount
+
+
+class CooperatorPortal(PortalAccount):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # See cooportal_portal/controllers/portal.py for why this code is the
+        # way it is.
+        if "OPTIONAL_BILLING_FIELDS" not in vars(self):
+            self.OPTIONAL_BILLING_FIELDS = PortalAccount.OPTIONAL_BILLING_FIELDS.copy()
+        self.OPTIONAL_BILLING_FIELDS.extend(["national_number"])
+
+    def _prepare_portal_layout_values(self):
+        values = super()._prepare_portal_layout_values()
+        national_number = request.env.user.partner_id.get_belgian_national_number()
+        values.update({"national_number": national_number})
```

Comment:
why this syntax instead of the more common (and apparently faster) `values["national_number"] = national_number`?

Reply:
No good reason. Was copied from portal. I changed it, but it doesn't make a difference.